### PR TITLE
Start running some tests in release config.

### PR DESCRIPTION
**Pull request by mbrandonw**
In order to workaround some Swift bugs we sometimes conditionally compile code for debug vs. release, and we currently have a blindspot for running tests in release mode. I've done the simplest thing so far by wrapping a bunch of tests that are not release-ready in `#if DEBUG`, and we can slowly starting moving them over to be release-friendly as we have time.

It's best to look at the diff with whitespace turned off.

### CODE REVIEW

File: Sources/ComposableArchitecture/Store.swift [hunk 1]
```diff
@@ -581,6 +581,7 @@ public typealias StoreOf<R: ReducerProtocol> = Store<R.State, R.Action>
   >: ReducerProtocol {
     let rootStore: Store<RootState, RootAction>
     let toScopedState: (RootState) -> ScopedState
+    private var parentStores: [Any]
```
Comment (mbrandonw) [inline Sources/ComposableArchitecture/Store.swift:584]: This is @tgrapperon's fix for #1379. It fixes the test in Release_StoreTests.swift.

File: Examples/Todos/Todos.xcodeproj/xcshareddata/xcschemes/Todos.xcscheme [hunk 2]
```diff
@@ -41,7 +41,7 @@
       </Testables>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
```
Comment (stephencelis) [inline Examples/Todos/Todos.xcodeproj/xcshareddata/xcschemes/Todos.xcscheme:44]: Should this change be reverted?

File: Tests/ComposableArchitectureTests/EffectRunTests.swift [hunk 3]
```diff
@@ -42,35 +41,37 @@ final class EffectRunTests: XCTestCase {
     await store.receive(.response)
   }
 
-  func testRunUnhandledFailure() async {
-    XCTExpectFailure(nil, enabled: nil, strict: nil) {
-      $0.compactDescription == """
-        An 'Effect.run' returned from "ComposableArchitectureTests/EffectRunTests.swift:62" threw \
-        an unhandled error. …
+  #if DEBUG
+    func testRunUnhandledFailure() async {
```
Comment (mbrandonw) [inline Tests/ComposableArchitectureTests/EffectRunTests.swift:45]: None of the runtime warnings / test failure tests can run in release, so maybe we just move all such tests to their own files eventually.

**Review by stephencelis — APPROVED** [review]

File: Tests/ComposableArchitectureTests/Release_StoreTests.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import Combine
+import ComposableArchitecture
+import XCTest
+
+final class Release_StoreTests: XCTestCase {
```
Comment (mbrandonw) [inline Tests/ComposableArchitectureTests/Release_StoreTests.swift:5]: this test demonstrates the behavior noticed in #1379

File: Examples/Todos/Todos.xcodeproj/xcshareddata/xcschemes/Todos.xcscheme (hunk 2 above)
Comment (mbrandonw) [reply Examples/Todos/Todos.xcodeproj/xcshareddata/xcschemes/Todos.xcscheme]: yep done

File: Sources/ComposableArchitecture/Store.swift (hunk 1 above)
Comment (tgrapperon) [inline Sources/ComposableArchitecture/Store.swift:584]:
```suggestion
    private let parentStores: [Any]
```

**Pull request merged**